### DISCUSSION TITLE
Proposal to Extend Sunbird InQuiry Spec to accomodate exam like assessments with server side response processing

### DISCUSSION THREAD

**Discussion by sharathkashyap** [Sunbird-inQuiry/Community#74] [Feature Requests]
Hello Team,
           I hereby propose to extend the Sunbird InQuiry Specification to accomodate exam like assessments with server side response processing capability.
           
**Need**:
For OGHR, Platform needs to enable user to undertake exams. Platform should be capable of processing the user responses and evaluate these responses with relevant applicable scores. Outcome of this design will enable the platform users to take up examination with server side evaluation of results.

**Design Problems** :
Allow the response processing of question to be done on server rather than client (QuML Player) as happening today.
Scalable Response Processing Solution for Question Set Evaluation on Server.
Score Calculation Based on Content State Update.
Solution to the question needs to be masqueraded or excluded from the Question Read API/ Question List API.

Two kinds of Response Processing is considered for the tech design:
          Entire Question Set Response Processing (Tech Design Currently Proposes extension here)
          Question by Question Response Processing (Futuristic not considered in the design)

Current System:
![image](https://user-images.githubusercontent.com/19199471/234530486-63a3889a-590c-41e1-a769-3ff6b8b77793.png)
Proposed System:
![image](https://user-images.githubusercontent.com/19199471/234530637-44bf18d4-e62c-40f7-9d58-bed2547f1a34.png)

The Tech Design is still currently Work In Progress. We will be requesting for design review in next week. Kindly go through the detailed confluence link for more updates.

https://project-sunbird.atlassian.net/wiki/spaces/CON/pages/3304685658/Question+Set+Response+Processing+-+Exam

cc: @surendrasinghs @sajeshkayyath @krgauraw

**Comment by sajeshkayyath** [comment]
Thanks for sharing the details @sharathkashyap 

cc  @PoojaJManjooran / @kameshbhr / @surendrasinghs / @krgauraw / @rajnishdargan / @vaibhavbhuva / @pallakartheekreddy
